### PR TITLE
[#21] Integrate survey detail to the UI

### LINKED ISSUE #21: [Integrate] As a user, I can see a survey detail
## Why
After fetching detail from the API and having the UI ready, start binding the survey detail accordingly.
 
## Acceptance Criteria

Display this screen with the actual data:

image

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #21

## What happened 👀

Integrate the detail info a Survey to the Survey page after fetching the data from the graphql API on the last PR.
 
## Insight 📝
- Just binding data, update the holder, and prepare for the next integration (go to the question, show shimmer etc...)
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/13435717/124589626-69fa6900-de84-11eb-95bc-099b2a082ab1.mov

### CODE REVIEW

File: lib/pages/survey/survey_controller.dart [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
   void onInit() {
     super.onInit();
 
     final surveyId = Get.arguments[DATA_SURVEY_ID];
     // TODO: remove this when integrate
     final getSurveyUseCase = Get.find<GetSurveyDetailUseCase>();
-    getSurveyUseCase.call(surveyId).then((value) => print("Value: $value"));
+    getSurveyUseCase.call(surveyId).then((result) => {
+          if (result is Success<Survey>)
+            {_survey.value = Optional.of(result.value)}
```
Comment (manh-t) [inline lib/pages/survey/survey_controller.dart:24]: Could you please re-format here a bit? 😂

File: lib/pages/survey/survey_page.dart [hunk 2]
```diff
@@ -1,92 +1,108 @@
...
-                image: DecorationImage(
-                  image: NetworkImage(
-                      "https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_l"),
-                  fit: BoxFit.cover,
-                  colorFilter: ColorFilter.mode(
-                      Colors.black.withOpacity(0.5), BlendMode.overlay),
+        builder: (controller) => GetX<SurveyController>(
+          builder: (state) {
+            return state.optionalSurvey.isPresent
+                ? SurveyDetail(state: state)
```
Comment (manh-t) [inline lib/pages/survey/survey_page.dart:18]: I think putting `survey` is enough. How do you think? 😄 
```suggestion
                ? SurveyDetail(survey: state.optionalSurvey.value)
```

Comment (sleepylee) [reply lib/pages/survey/survey_page.dart:18]: oh yes 👌

File: lib/pages/survey/survey_controller.dart (hunk 1 above)
Comment (sleepylee) [reply lib/pages/survey/survey_controller.dart:24]: hmmm the auto-format turned it into this...

**Review by manh-t — APPROVED** [review]
LGTM 👍

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Pull request merged**